effectful-core/src/Effectful/Exception.hs

```diff
@@ -1,87 +1,112 @@
--- | The 'Eff' monad comes with instances of 'MonadThrow', 'MonadCatch' and
--- 'MonadMask' from the
--- [@exceptions@](https://hackage.haskell.org/package/exceptions) library, so
--- this module simply re-exports the interface of the
--- [@safe-exceptions@](https://hackage.haskell.org/package/safe-exceptions)
--- library.
---
--- Why @safe-exceptions@ and not @exceptions@? Because the former makes it much
--- easier to correctly deal with asynchronous exceptions (for more information
--- see its [README](https://github.com/fpco/safe-exceptions#readme)) and
--- provides more convenience functions.
+{-# LANGUAGE CPP #-}
+-- | Support for runtime exceptions.
+--
+-- This module provides thin wrappers over functions from "Control.Exception" as
+-- well as several utility functions for convenience.
+--
+-- However, the 'Eff' monad provides instances for 'C.MonadThrow',
+-- 'C.MonadCatch' and 'C.MonadMask', so you can also use the following
+-- libraries:
+--
+-- - [@exceptions@](https://hackage.haskell.org/package/exceptions)
```

Comment:
While I see the point of using `safe-` and `annotated-` variants, is there a concrete recommendation that the documentation could make for new adopters of `effectful`? Like:

> Start with using the functions of this module, and if <conditions> then switch to `exceptions`.

What would motivate someone to use the implementations in `exceptions`?

Reply:
Yeah, I removed this, see https://github.com/haskell-effectful/effectful/pull/257#discussion_r1802865352

Reply:
Okay, I see!